fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -2732,8 +2735,14 @@ private static void validateIndexIntegrity(Index index, @Nonnull Subspace subspa
             if (FDBDirectory.isEntriesFile(file) || FDBDirectory.isSegmentInfo(file) || FDBDirectory.isFieldInfoFile(file)
                     || file.endsWith(".pky")) {
                 assertFalse(fileReference.getContent().isEmpty(), "fileName=" + file);
+            } else if (FDBDirectory.isStoredFieldsFile(file) && (segmentName != null)) {
+                // It is OK for a stored fields info for the next segment to be created at this point.
```

Comment:
This seems somewhat supsicious, I'm not sure why this would be.

Reply:
There is one test in particular that causes (simpleInsertDeleteAndSearch) `_1.fsf` to be present when validating the `_0` segment.

Reply:
ok, I think I might have a very vague understanding of how that could happen....